code-challenges.js

```diff
@@ -1,34 +1,112 @@
+
+// var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+// // Expected output: [-9, 9, 7, 199]
+// var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+// // Expected output: [-823, 7, 23]
+
+//draft up mock test w/Jest 
+// describe("oddsInOrder", () => {
+//     test("takes in an array and returns a new array of only odd numbers sorted from least to greatest", () => {
+//         expect(oddsInOrder(fullArr1)).toEqual([-9, 7, 9, 199])
+//         expect(oddsInOrder(fullArr2)).toEqual([-823, 7, 23])
+//     })
+// })
+
+// //Create function that takes in array as an argument
+// const oddsInOrder = (arr) => {
+//     //Filter through array if value equals/truthy to an odd number it is automatically pushed/returned to new array
+//     //sort new array in ascending order
+//     return arr.filter(value => typeof value === "number" && value % 2 !== 0).sort((a, b) => a -b)
+// }
+// //console.log to ensure desired results
+// console.log(oddsInOrder(fullArr1));
+// console.log(oddsInOrder(fullArr2))
 
 
 // --------------------3) Create a function that takes in a string of a single word and returns the middle letter of the word. If the word is an even number of letters, return the two middle letters.
 
-var middleLetters1 = "hello"
-// Expected output: “l”
-var middleLetters2 = "rhinoceros"
-// Expected output: “oc”
-
-
+// var middleLetters1 = "hello"
+// // Expected output: “l”
+// var middleLetters2 = "rhinoceros"
+// // Expected output: “oc”
+
+//draft up mock test w/Jest 
+// describe("middleLetter", () => {
+//     test("takes in a string of a single word and returns the middle letter of the word. If the word is an even number of letters, return the two middle letters", () => {
+//         expect(middleLetter(middleLetters1)).toEqual("l")
+//         expect(oddsInOrder(middleLetters2)).toEqual("oc")
+//     })
+// })
+
+// declare function that takes in a string
+// const middleLetter = (str) => {
+//     //return a new/substring of the middle letter. If there are even numbers return two middle letters
+//     //Look for higher order functions I could use ".substr"
+//     //Remember that odd numbers do not return whol numbers will need to use Math.round() for any potential issues there.
+//     //Try and use ternary operations rather than if else statements
+//     return str.substr(Math.round(str.length / 2), str.length % 2 === 0 ? 2
+//     // str.length % 2 === 0(condition) ?(is it true?) 2(then print a string of 2) :(if false) 1(return a string of 1)
+//     : 1)
+// }
+//  console.log(middleLetter(middleLetters1))
+//  console.log(middleLetter(middleLetters2))
 
 
 // --------------------4) Create a CLASS to get the area of a sphere. Create three spheres with different radi as test cases. Area of a sphere =  4πr^2 (four pi r squared)
 
-
+//not really sure how to set up a test for this sort of problem
```

Comment:
fair. it's a bit more complex. once we get into testing with enzyme testing in react, class testing will be a little easier.